rocketmq-spring-boot-starter/src/main/java/org/apache/rocketmq/spring/starter/RocketMQTransactionAnnotationProcessor.java

```diff
@@ -0,0 +1,94 @@
+import org.apache.rocketmq.spring.starter.annotation.RocketMQTransactionListener;
+import org.apache.rocketmq.spring.starter.config.TransactionHandler;
+import org.apache.rocketmq.spring.starter.config.TransactionHandlerRegistry;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.*;
+import org.springframework.context.expression.StandardBeanExpressionResolver;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.AnnotationUtils;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+public class RocketMQTransactionAnnotationProcessor
+    implements BeanPostProcessor, Ordered, BeanFactoryAware, SmartInitializingSingleton {
+  private BeanFactory beanFactory;
+  private BeanExpressionResolver resolver = new StandardBeanExpressionResolver();
+  private final Set<Class<?>> nonAnnotatedClasses =
+      Collections.newSetFromMap(new ConcurrentHashMap<Class<?>, Boolean>(64));
+
+  @Autowired(required = false)
+  private TransactionHandlerRegistry transactionHandlerRegistry;
+
+
+  @Override
+  public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
+    this.beanFactory = beanFactory;
+    if (beanFactory instanceof ConfigurableListableBeanFactory) {
+      this.resolver = ((ConfigurableListableBeanFactory) beanFactory).getBeanExpressionResolver();
+    }
+  }
+
+  @Override
+  public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+    return bean;
+  }
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+    if (!this.nonAnnotatedClasses.contains(bean.getClass())) {
+      Class<?> targetClass = AopUtils.getTargetClass(bean);
+      RocketMQTransactionListener listener = AnnotationUtils.findAnnotation(targetClass, RocketMQTransactionListener.class);;
+      this.nonAnnotatedClasses.add(bean.getClass());
+      if (listener == null) { //for quick search
+          log.trace("No @RocketMQTransactionListener annotations found on bean type: {}", bean.getClass());
+      }else {
+            try{
+              processTransactionListenerAnnotation(listener, bean, beanName);
+            } catch (MQClientException e) {
+              log.error("failed to process annotation " + listener, e);
+              throw new BeanCreationException("failed to process annotation " + listener, e);
+            }
+          }
+    }
+
+    return bean;
+  }
+
+  private void processTransactionListenerAnnotation(RocketMQTransactionListener anno, Object bean, String beanName) throws MQClientException {
+    if (transactionHandlerRegistry == null) {
+      throw new MQClientException(-1, "Bad usage of @RocketMQTransactionListener, the class must work with producer rocketMQTemplate");
```

Comment:
It is recommended to define an explicit exception error code.

Reply:
The org.apache.rocketmq.common.protocol.ResponseCode is defined in main project (apache/rocketmq) rather than the project external  project, so it is difficult to define explicit value. So I used the generic code -1.

Reply:
will change as "new MQClientException(..., null)"